c7n/actions.py

```diff
@@ -522,3 +571,102 @@ def process(self, resources, event):
             {'key': self.data['tag'], 'value': user},
             self.manager).process(untagged)
         return untagged
+
+
+class PutMetric(BaseAction):
+    """Action to put metrics based on an expression into CloudWatch metrics
+    
+    :example:
+    
+        .. code-block: yaml
+        
+            policies:
+              - name: track-attached-ebs
+                resource: ec2
+                comment: |
+                  Put the count of the number of EBS attached disks to an instance
+                filters:
+                  - Name: tracked-ec2-instance
+                actions:
+                  - type: put-metric
+                    key: Reservations[].Instances[].BlockDeviceMappings[].DeviceName
+                    namespace: Usage Metrics
+                    metric_name: Attached Disks
+                    op: count
+                    units: Files
+
+    op and units are optional and will default to simple Counts.
+    """
+    permissions = {'cloudwatch:PutMetricData',} #  typically lowercase servicename:TitleCaseActionName
```

Comment:
minor style stuff, worth running through flake8 or integrating with your editor, i think it would flag the comment not two spaces after code, i'd just stick this above the line to conform to 80 char lines